src/bpm/commands/list.go

```diff
@@ -36,18 +39,62 @@ var listCommandCommand = &cobra.Command{
 func listContainers(cmd *cobra.Command, _ []string) error {
 	cmd.SilenceUsage = true
 
+	processes := []*models.Process{}
+	for _, job := range bosh.JobNames() {
+		bpmCfg := config.NewBPMConfig(bosh.Root(), job, "")
```

Comment:
We don't have to address this in this commit but it feels bad to create a configuration with an empty process name to get at some of its functionality. This configuration can't answer questions about processes which is surprising based on its type.

I think we can split the configuration into two separate objects and remove the third argument from this. One for the job configuration and one for the process configuration. Maybe there can be a function on the job configuration to get the related process configuration.

```go
jobCfg := config.NewBPM(bosh.Root(), jobName)
procCfg := jobCfg.ProcessConfig(processName)
```

Reply:
I felt the same way but I wasn't quite sure how to tackle it yet. The config objects seem tangled right now.

`BPMConfig` has all of the paths and container IDs for a given set of arguments. I see that we can probably separate job independent paths from paths that require the process name, but we would then have to pass both objects around when constructing the runc bundle and interacting with the BOSH directory structure.

The current `JobConfig` and `ProcessConfig` are the actual structure of the `bpm.yml`. I wanted to keep these structs the way they are now. I wonder if they have a more accurate name? Either way this was another reason why I punted on splitting the config.

The `BOSHConfig` is agnostic of processes and jobs.

Reply:
I wonder if we can leave things on the same object but pass in the process name to the functions which require them rather than it being in the constructor?

Either way, it doesn't need to happen as part of this change.